src/preprocessing/passes/daneshvar.cpp

```diff
@@ -0,0 +1,650 @@
+
+    // Map to keep track of visited nodes
+    std::unordered_map<Node, bool> visited;
+
+    // Initialize a global variable counter for bound variables
+    static int globalVarCounter = 0;
+
+    // Push the root node onto the stack
+    stack.push(n);
+
+    while (!stack.empty())
+    {
+        Node current = stack.top();
+
+        auto [it, inserted] = visited.emplace(current, false);
+        if (inserted)
+        {
+            // First time seeing this node
+
+            if (current.isConst() || current.isVar())
+            {
+                // For constants and variables, process immediately
+                if (current.isVar())
+                {
+                    if (current.getKind() == cvc5::internal::Kind::BOUND_VARIABLE)
+                    {
+                        // Handle bound variable
+                        auto it_bv = boundVar2node.find(current);
+                        if (it_bv != boundVar2node.end())
+                        {
+                            normalized[current] = it_bv->second;
+                        }
+                        else
+                        {
+                            int id = globalVarCounter++;
+                            std::string new_var_name =
+                                "u" + std::string(8 - numDigits(id), '0') + std::to_string(id);
+                            Node ret = nodeManager->mkBoundVar(new_var_name, current.getType());
+                            boundVar2node[current] = ret;
+                            normalized[current] = ret;
+                        }
+                    }
+                    else
+                    {
+                        // Handle free variable
+                        auto varName = current.toString();
+                        auto it_fv = freeVar2node.find(varName);
+                        if (it_fv != freeVar2node.end())
+                        {
+                            normalized[current] = it_fv->second;
+                        }
+                        else
+                        {
+                            std::vector<Node> cnodes;
+                            int id = freeVar2node.size();
+                            std::string new_var_name =
+                                "v" + std::string(8 - numDigits(id), '0') + std::to_string(id);
+                            cnodes.push_back(nodeManager->mkConst(String(new_var_name, false)));
+                            Node gt = nodeManager->mkConst(SortToTerm(current.getType()));
+                            cnodes.push_back(gt);
+                            Node ret = nodeManager->getSkolemManager()->mkSkolemFunction(
+                                SkolemFunId::INPUT_VARIABLE, cnodes);
+                            freeVar2node[varName] = ret;
+                            normalized[current] = ret;
+                            d_preprocContext->addSubstitution(current, ret);
+                        }
+                    }
```

Comment:
This code is still mapping symbols to nodes.

Reply:
Fixed